tests/unit/utils/test_wait_for_service.py

```diff
@@ -0,0 +1,22 @@
+from src.utils.wait_for_service import wait_for_service
+import pytest
+import logging
+import time
+
+
+def test_init_config_invalid_config_url(caplog):
+    search2_logger = logging.getLogger('search2')
+    # This ensures that logs are propagated and can be captured by caplog
+    search2_logger.propagate = True
+    with caplog.at_level(logging.INFO, logger='search2'):
+        with pytest.raises(SystemExit) as se:
+            wait_for_service('https://foo.bar.baz', 'foo', timeout=0)
+        start = time.time()
+        assert se.type == SystemExit
+        assert se.value.code == 1
+        # If it took less than 10 seconds, we triggered the exist on timeout
+        # on the first check.
```

Comment:
Seems a tad bit hacky but I guess it is fine.  I'm not sure I have an alternative.

Reply:
You mean forcing the exit with a short timeout?

The basic idea is that an error in the  call to monitor the service url will result in a nearly immediate return when supplying a short timeout, and in that test I didn't want to be concerned with the actual  timeout behavior. 

In the coverage reports, the success branch was already covered, the timeout wait loop covered, but the exit was not. That was what I was trying to guide the code through, minimally. The minimal pathway  through would avoid the timeout wait loop.

Since there were no unit tests for this code, I would bet that the reason it had coverage at all was because the unit tests actually use that function to spin up the server.

I've updated the tests to include timeout conditions.